lib/accountregistry.h

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: 2020 Kitsune Ral <Kitsune-Ral@users.sf.net>
+// SPDX-FileCopyrightText: Tobias Fella <fella@posteo.de>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <QtCore/QObject>
+#include <QtCore/QList>
+#include <QtCore/QAbstractListModel>
+
+namespace Quotient {
+class Connection;
+
+class AccountRegistry : public QAbstractListModel {
+    Q_OBJECT
+public:
+    using const_iterator = QMap<QString, Connection *>::const_iterator;
+
+    enum EventRoles {
+        ConnectionRole = Qt::UserRole + 1,
+    };
+
+    static AccountRegistry &instance() {
+        static AccountRegistry _instance;
+        return _instance;
+    }
+
+    const QList<Connection*> accounts() const;
+    void add(Connection* a);
+    void drop(Connection* a);
+    bool isLoggedIn(const QString& userId) const;
+    Connection* operator[](int);
+    Connection* operator[](int) const;
+    bool isEmpty() const;
+    int count() const;
+    bool contains(Connection*) const;
+    Connection* get(const QString& userId);
+
+    const_iterator cbegin() const;
+    const_iterator cend() const;
+    const_iterator begin() const;
+    const_iterator end() const;
+
+    [[nodiscard]] QVariant data(const QModelIndex &index, int role = ConnectionRole) const override;
+    [[nodiscard]] int rowCount(const QModelIndex &parent = QModelIndex()) const override;
+
+    [[nodiscard]] QHash<int, QByteArray> roleNames() const override;
+
+private:
+    AccountRegistry();
+
+    QMap<QString, Connection *> m_accounts;
+
+signals:
+    void addedAccount(Connection* a);
+    void aboutToDropAccount(Connection* a);
```

Comment:
As you derive from `QAbstractListModel`, I don't think it's worth replicating signals that serve the same function (even if they provide less simplistic API). Quaternion will do fine with connecting to `rowsInserted()` and `aboutToRemoveRows()` instead.